source/detail/parser/x86_64.cpp

```diff
@@ -369,13 +365,12 @@ namespace smeagle::x86_64 {
 
           // Otherwise we generate a string from register classes
         } else {
-          location = getRegistersString(regClasses, order);
+          location = locator.getRegistersString(regClasses);
```

Comment:
What happens when `getRegisterString` returns "stack" to signify that we've run out of registers?

Reply:
I have no idea.

Reply:
I thought when we run out it just gets stored in memory (e.g., stack).

Reply:
It does, but we still need to make a spot for it relative to the framebase-- just like in the MEMORY case.

Reply:
Hmm we should probably have the allocator control that too then.

Reply:
Yup. I would consider using a different allocator for the stack and registers so you can keep the interfaces nice and simple.

Reply:
I already added it to our main allocator. I think they need to be together because when we run out of registers we have to return a framebase too.

Reply:
That's a perfectly fine implementation. I like making lots of little types when I write C++, but there's no "right way" to do it!

Reply:
How would you go about it? Another framebase allocator that the RegisterAllocator uses? Or have them both at the bottom, and the Framebas eallocator is used given the return of "stack"?

Reply:
okay I gave it a shot - since we need to determine framebases for two different registers and I would need to check if the std::optional string is set or not, I opted to make the FramebaseAllocator and have it be part of the RegisterAllocator class. If we needed to separate them we would need to iterate through the pair at the bottom and then handle each case separately, which I didn't like as much. This way, we can use the FramebaseAllocator alongside the RegisterAllocator, and it could be used in other contexts as needed. Or in other words, I don't see any cases where a RegisterAllocator could work without a FramebaseAllocator, but the opposite isn't true.

Reply:
I actually like this implementation more than the one I was thinking of because we can still have no logical at the call site which is a big plus. My version would be more flexible, but we don't need that much re-use here. Nice!